### PR TITLE
CNV-22036: Update CLI, UI uninstall procedures

### PR DESCRIPTION
**Pull request by apinnick**
Version(s): 4.11+

Issue: [CNV-22036](https://issues.redhat.com/browse/CNV-22036)

Link to docs preview: [Uninstalling OpenShift Virtualization](https://52807--docspreview.netlify.app/openshift-enterprise/latest/virt/install/uninstalling-virt.html)

QE review: @OhadRevah 
- [x] QE has approved this change.

Additional information: This PR consolidates two assemblies into one.

### CODE REVIEW

File: virt/install/uninstalling-virt.adoc [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+toc::[]
+
+You can uninstall {VirtProductName} by using the {product-title} xref:../../web_console/web-console.adoc#web-console-overview_web-console[web console] or the command line interface (xref:../../cli_reference/openshift_cli/getting-started-cli.adoc#cli-getting-started[CLI]).
+
+[id='uninstalling-virt-web-console_{context}']
+== Uninstalling {VirtProductName} using the web console
+
+You uninstall {VirtProductName} by deleting the {VirtProductName} Operator from the cluster and then deleting the `openshift-cnv` namespace.
+
+include::modules/olm-deleting-operators-from-a-cluster-using-web-console.adoc[leveloffset=+2]
```
Comment (tiraboschi) [inline virt/install/uninstalling-virt.adoc:16]: In our case we still recommend to delete the HyperConverged custom resource (CR) first,
for simpler operators this can be automatically done on operator removal but this is not our case.

**Review by tiraboschi — CHANGES_REQUESTED** [review]

File: virt/install/uninstalling-virt.adoc [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+:context: uninstalling-virt
+
+toc::[]
+
+You can uninstall {VirtProductName} by using the {product-title} xref:../../web_console/web-console.adoc#web-console-overview_web-console[web console] or the command line interface (xref:../../cli_reference/openshift_cli/getting-started-cli.adoc#cli-getting-started[CLI]).
+
+[id='uninstalling-virt-web-console_{context}']
+== Uninstalling {VirtProductName} using the web console
+
+You uninstall {VirtProductName} by deleting the {VirtProductName} Operator from the cluster and then deleting the `openshift-cnv` namespace.
```
[the remark this answers is not in the document]
Comment (apinnick) [reply virt/install/uninstalling-virt.adoc:14]: When Ohad Revah deleted the HyperConverged CR, the Operator kept reconciling it. When he deleted the Operator, it deleted the CR as well. 

Is there a reason why we need the steps to be separate?

File: virt/install/uninstalling-virt.adoc (hunk 1 above)
Comment (apinnick) [reply virt/install/uninstalling-virt.adoc:16]: Won't the HCO reconcile process be an issue if the user tries to delete the HyperConverged CR first? Same question as in your previous comment.

**Comment by apinnick** [comment]
@OhadRevah This PR is ready for QE review. Thanks!

**Review by tiraboschi — APPROVED** [review]

**Review by ohadrevach — APPROVED** [review]

File: virt/install/uninstalling-virt.adoc [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+[IMPORTANT]
+====
+You must delete all xref:../../virt/virtual_machines/virt-delete-vms.adoc#virt-delete-vm-web_virt-delete-vms[virtual machines], xref:../../virt/virtual_machines/virt-manage-vmis.adoc#virt-deleting-vmis-cli_virt-manage-vmis[virtual machine instances],
+and xref:../../virt/virtual_machines/virtual_disks/virt-deleting-datavolumes.adoc#virt-deleting-dvs_virt-deleting-datavolumes[data volumes] before you uninstall {VirtProductName}. If {VirtProductName} workloads exist, you cannot delete the `HyperConverged` custom resource (CR).
+====
+
+[id='uninstalling-virt-web-console_{context}']
+== Uninstalling {VirtProductName} using the web console
+
+You uninstall {VirtProductName} by deleting the `HyperConverged` CR, the {VirtProductName} Operator, and the `openshift-cnv` namespace.
```
Comment (sjhala-ccs) [inline virt/install/uninstalling-virt.adoc:20]: Because the user needs to perform these tasks in sequence, I would suggest rewriting this as:
````
You uninstall {VirtProductName} by performing the following tasks:

1. Delete the HyperConverged custom resource (link to procedure module)
2. Delete the {VirtProductName} Operator (link to procedure module)
3. Delete the `openshift-cnv` namespace (link to procedure module)
````

Also see my comment for lines 24 and 28

File: virt/install/uninstalling-virt.adoc [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+You can uninstall {VirtProductName} by using the {product-title} xref:../../web_console/web-console.adoc#web-console-overview_web-console[web console] or the command line interface (xref:../../cli_reference/openshift_cli/getting-started-cli.adoc#cli-getting-started[CLI]).
+
+[IMPORTANT]
+====
+You must delete all xref:../../virt/virtual_machines/virt-delete-vms.adoc#virt-delete-vm-web_virt-delete-vms[virtual machines], xref:../../virt/virtual_machines/virt-manage-vmis.adoc#virt-deleting-vmis-cli_virt-manage-vmis[virtual machine instances],
+and xref:../../virt/virtual_machines/virtual_disks/virt-deleting-datavolumes.adoc#virt-deleting-dvs_virt-deleting-datavolumes[data volumes] before you uninstall {VirtProductName}. If {VirtProductName} workloads exist, you cannot delete the `HyperConverged` custom resource (CR).
+====
+
+[id='uninstalling-virt-web-console_{context}']
+== Uninstalling {VirtProductName} using the web console
```
Comment (mburke5678) [inline virt/install/uninstalling-virt.adoc:18]: s/using/by using

File: virt/install/uninstalling-virt.adoc [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+
+You uninstall {VirtProductName} by deleting the `HyperConverged` CR, the {VirtProductName} Operator, and the `openshift-cnv` namespace.
+
+include::modules/virt-deleting-deployment-custom-resource.adoc[leveloffset=+2]
+
+Delete the {VirtProductName} Operator from the `openshift-cnv` namespace.
+
+include::modules/olm-deleting-operators-from-a-cluster-using-web-console.adoc[leveloffset=+2]
```
Comment (mburke5678) [inline virt/install/uninstalling-virt.adoc:26]: s/using/by using in the `olm-deleting-operators-from-a-cluster-using-web-console`file, if you would be so kind.

File: virt/install/uninstalling-virt.adoc [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+
+Delete the `openshift-cnv` namespace.
+
+include::modules/deleting-a-namespace-using-the-web-console.adoc[leveloffset=+2]
```
Comment (mburke5678) [inline virt/install/uninstalling-virt.adoc:30]: s/using/by using in the `deleting-a-namespace-using-the-web-console` file, if you would be so kind.

File: virt/install/uninstalling-virt.adoc [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+
+You uninstall {VirtProductName} by deleting the `HyperConverged` CR, the {VirtProductName} Operator, and the `openshift-cnv` namespace.
+
+include::modules/virt-deleting-deployment-custom-resource.adoc[leveloffset=+2]
+
+Delete the {VirtProductName} Operator from the `openshift-cnv` namespace.
+
+include::modules/olm-deleting-operators-from-a-cluster-using-web-console.adoc[leveloffset=+2]
+
+Delete the `openshift-cnv` namespace.
```
Comment (mburke5678) [inline virt/install/uninstalling-virt.adoc:28]: Same comment here as for _Delete the {VirtProductName} Operator from the `openshift-cnv` namespace._

File: virt/install/uninstalling-virt.adoc [hunk 8]
```diff
@@ -0,0 +1,32 @@
+:_content-type: ASSEMBLY
+[id="uninstalling-virt"]
+include::_attributes/common-attributes.adoc[]
+= Uninstalling {VirtProductName}
+:context: uninstalling-virt
+
+toc::[]
+
+You can uninstall {VirtProductName} by using the {product-title} xref:../../web_console/web-console.adoc#web-console-overview_web-console[web console] or the command line interface (xref:../../cli_reference/openshift_cli/getting-started-cli.adoc#cli-getting-started[CLI]).
```
Comment (mburke5678) [inline virt/install/uninstalling-virt.adoc:9]: Maybe this is a virt convention, but in uninstall modules, I tend to link the _you can uninstall by using the web console or CLI_ to the uninstall sections. Feel free to ignore.

File: virt/install/uninstalling-virt.adoc [hunk 9]
```diff
@@ -0,0 +1,32 @@
...
+====
+
+[id='uninstalling-virt-web-console_{context}']
+== Uninstalling {VirtProductName} using the web console
+
+You uninstall {VirtProductName} by deleting the `HyperConverged` CR, the {VirtProductName} Operator, and the `openshift-cnv` namespace.
+
+include::modules/virt-deleting-deployment-custom-resource.adoc[leveloffset=+2]
+
+Delete the {VirtProductName} Operator from the `openshift-cnv` namespace.
```
Comment (sjhala-ccs) [inline virt/install/uninstalling-virt.adoc:24]: Suggest putting this sentence under a **Next steps** heading with the `.` formatting. Also applies to line 28.

File: modules/virt-deleting-virt-cli.adoc [hunk 10]
```diff
@@ -4,45 +4,39 @@
...
-. Delete the subscription of the {VirtProductName} operator in the Operator Lifecycle Manager (OLM):
+. Delete the {VirtProductName} Operator subscription:
 +
 [source,terminal]
 ----
 $ oc delete subscription kubevirt-hyperconverged -n openshift-cnv
 ----
 
-. Set the cluster service version (CSV) name for {VirtProductName} as an environment variable:
+. Set the {VirtProductName} Operator CSV environment variable:
```
Comment (mburke5678) [inline modules/virt-deleting-virt-cli.adoc:32]: I see this is existing text, but it seems a bit much to have folks set a env var for an operator they are going to remove. Maybe just get the CSV name and `oc delete csv {csv_name} -n openshift-cnv`. But that's likely just a me thing.

**Comment by mburke5678** [comment]
@apinnick A few nits. Otherwise LGTM.

File: modules/virt-deleting-virt-cli.adoc [hunk 11]
```diff
@@ -4,45 +4,39 @@
...
-
-- Access to a {VirtProductName} cluster using an account with
-`cluster-admin` permissions.
-
-[NOTE]
-====
-When you delete the subscription of the {VirtProductName} operator in the OLM by using the CLI, the `ClusterServiceVersion` (CSV) object is not deleted from the cluster. To completely uninstall {VirtProductName}, you must explicitly delete the CSV.
-====
+- Access to an {product-title} cluster using an account with `cluster-admin` permissions.
+- `oc` command installed on workstation.
```
Comment (sjhala-ccs) [inline modules/virt-deleting-virt-cli.adoc:14]: I have seen this written as `You have installed the OpenShift CLI (oc).` in most of our existing procedures. Suggest rewording it for consistency.

**Review by sjhala-ccs — COMMENTED** [review]
Just a few suggestions for you to consider, otherwise LGTM

File: modules/virt-deleting-virt-cli.adoc (hunk 11 above)
Comment (mburke5678) [reply modules/virt-deleting-virt-cli.adoc:14]: I did see this format used elsewhere in the docs. For example, in the Operators book, [Deleting Operators from a cluster using the CLI](https://docs.openshift.com/container-platform/4.11/operators/admin/olm-deleting-operators-from-cluster.html#olm-deleting-operator-from-a-cluster-using-cli_olm-deleting-operators-from-a-cluster).
I do think Shikha's suggestion is used far more frequently.

File: virt/install/uninstalling-virt.adoc (hunk 4 above)
Comment (apinnick) [reply virt/install/uninstalling-virt.adoc:18]: Why are you reviewing a CNV PR? They are reviewed by our own TWs, in the @aligned-peer-review-squad. I set the "CNV" label, so it should not be picked up by the OpenShift peer review squad.

Comment (mburke5678) [reply virt/install/uninstalling-virt.adoc:18]: @apinnick This is the filter I am using. Perhaps it is out-dated? 

is:open label:peer-review-needed sort:updated-desc

File: modules/virt-deleting-deployment-custom-resource.adoc [hunk 12]
```diff
@@ -2,37 +2,20 @@
...
 .Procedure
 
-. From the {product-title} web console, select `openshift-cnv` from the *Projects* list.
-
 . Navigate to the *Operators* -> *Installed Operators* page.
 
-. Click *{VirtProductName}*.
+. Click the {VirtProductName} Operator.
 
 . Click the *{VirtProductName} Operator Deployment* tab.
```
[the remark this answers is not in the document]
Comment (apinnick) [reply modules/virt-deleting-deployment-custom-resource.adoc:17]: You're right. I saw that and forgot to make the correction.

File: modules/virt-deleting-virt-cli.adoc (hunk 11 above)
Comment (apinnick) [reply modules/virt-deleting-virt-cli.adoc:14]: OK. I copied it from an existing module (deleting CSV in the Operators section)

File: modules/virt-deleting-deployment-custom-resource.adoc [hunk 13]
```diff
@@ -2,37 +2,20 @@
...
+To uninstall {VirtProductName}, you first delete the `HyperConverged` custom resource (CR).
 
 .Procedure
 
-. From the {product-title} web console, select `openshift-cnv` from the *Projects* list.
-
 . Navigate to the *Operators* -> *Installed Operators* page.
 
-. Click *{VirtProductName}*.
+. Click the {VirtProductName} Operator.
```
[the remark this answers is not in the document]
Comment (apinnick) [reply modules/virt-deleting-deployment-custom-resource.adoc:15]: I'm not sure I see the difference between click and select. The Operator appears as a "card" and you have to click it. I admit that I did not check ISG because Patternfly seems to have a lot more UI elements than IBM uses.

I don't mind changing it but the logic is not clear to me. :-)

File: virt/install/uninstalling-virt.adoc (hunk 3 above)
Comment (tiraboschi) [reply virt/install/uninstalling-virt.adoc:20]: There is also another (optional?) step:
The OLM is not going to remove CRDs which are actually cluster scoped resources so neither removing the namespace will do.
In order to get rid of leftovers, product CRDs should be manually deleted.

**Review by tiraboschi — CHANGES_REQUESTED** [review]
I think we should also mention the removal of our CRDs

File: modules/virt-deleting-deployment-custom-resource.adoc (hunk 13 above)
[the remark this answers is not in the document]
Comment (sjhala-ccs) [reply modules/virt-deleting-deployment-custom-resource.adoc:15]: The way I see it the **Installed Operators** page has a list of Operators and the ISG recommends using _select_ to refer to selecting an item from a list. That was my logic :-)

[the remark this answers is not in the document]
Comment (apinnick) [reply modules/virt-deleting-deployment-custom-resource.adoc:15]: Really? I see squares or cards, not a list.

File: virt/install/uninstalling-virt.adoc (hunk 3 above)
Comment (apinnick) [reply virt/install/uninstalling-virt.adoc:20]: @tiraboschi I happened to see this [kubevirt-dev chat thread](https://groups.google.com/g/kubevirt-dev/c/68ae5_DIT-Q/m/8BtFvugqAwAJ) when I was looking for info on how to delete CRDs. Do I need to add the step that Chandler recommends?

File: modules/virt-deleting-deployment-custom-resource.adoc [hunk 14]
```diff
@@ -2,37 +2,20 @@
...
 [id="virt-deleting-deployment-custom-resource_{context}"]
-= Deleting the {VirtProductName} Operator Deployment custom resource
-
-To uninstall {VirtProductName}, you must first delete the *{VirtProductName} Operator Deployment* custom resource.
-
-.Prerequisites
+= Deleting the HyperConverged custom resource
 
-* Create the *{VirtProductName} Operator Deployment* custom resource.
+To uninstall {VirtProductName}, you first delete the `HyperConverged` custom resource (CR).
```
[the remark this answers is not in the document]
Comment (apinnick) [reply modules/virt-deleting-deployment-custom-resource.adoc:9]: I thought we had to use the 2nd person. Did that change?

File: modules/virt-deleting-virt-cli.adoc (hunk 10 above)
Comment (apinnick) [reply modules/virt-deleting-virt-cli.adoc:32]: They need the name and current version, not just the CSV name. This would be a better way to do it:

. Obtain the current CSV for the {VirtProductName} Operator:
$ oc get subscription kubevirt-hyperconverged -n openshift-cnv -o yaml | grep currentCSV

Comment (tiraboschi) [reply modules/virt-deleting-virt-cli.adoc:32]: you can simply identify it by label:
```
$ oc delete csv -n openshift-cnv -l operators.coreos.com/kubevirt-hyperconverged.openshift-cnv
```

File: virt/install/uninstalling-virt.adoc (hunk 3 above)
Comment (tiraboschi) [reply virt/install/uninstalling-virt.adoc]: also in this case you can simply query them by label:
```
$ oc delete crd --dry-run=client -l operators.coreos.com/kubevirt-hyperconverged.openshift-cnv
customresourcedefinition.apiextensions.k8s.io "cdis.cdi.kubevirt.io" deleted (dry run)
customresourcedefinition.apiextensions.k8s.io "hostpathprovisioners.hostpathprovisioner.kubevirt.io" deleted (dry run)
customresourcedefinition.apiextensions.k8s.io "hyperconvergeds.hco.kubevirt.io" deleted (dry run)
customresourcedefinition.apiextensions.k8s.io "kubevirts.kubevirt.io" deleted (dry run)
customresourcedefinition.apiextensions.k8s.io "networkaddonsconfigs.networkaddonsoperator.network.kubevirt.io" deleted (dry run)
customresourcedefinition.apiextensions.k8s.io "ssps.ssp.kubevirt.io" deleted (dry run)
customresourcedefinition.apiextensions.k8s.io "tektontasks.tektontasks.kubevirt.io" deleted (dry run)
```
remove the `--dry-run=client` option to make it effective.

Comment (apinnick) [reply virt/install/uninstalling-virt.adoc]: @tiraboschi The procedure is clear. Can the user do this only with the CLI? Or is there a way to do it in the UI?

File: virt/install/uninstalling-virt.adoc (hunk 5 above)
Comment (apinnick) [reply virt/install/uninstalling-virt.adoc:29]: I am not going to change the file name because it is used in other assemblies.

File: virt/install/uninstalling-virt.adoc (hunk 9 above)
Comment (apinnick) [reply virt/install/uninstalling-virt.adoc]: Sorry, I don't understand. Not sure how "Next steps" works.
